### ISSUE TITLE
BUG: Activation is offset (Sequence diagram, Teoz)

### ISSUE THREAD

**Issue by chipbite** [plantuml/plantuml#656]
This was reported previously: https://forum.plantuml.net/13569/activation-is-offset-using-teoz

Hello, 
Using teoz, for a message chain, where each participant gets activated, there is an offset, where only the first participant gets the lifeline drawn correctly (IMO), and the subsequent are drawn differently, with an offset relative to the first. Well, it is easier to see in a diagram, so check below ;-) ...

Code:
```
A -> B++: Get data
& A -> C++: 
& A -> D++: 
B -->> A--: Data
& C -->> A--: 
& D -->> A--:

```
**Current actual result for this:**
![image](https://user-images.githubusercontent.com/559199/132241126-2b48a8c0-beed-45ea-9417-f3032da3a32b.png)

**Actual / Generated result:**
![generated diagram](http://www.plantuml.com/plantuml/png/JO_DIiKm44RtynIV6pUNTEEkYzAVWBZN1uXcKWDD8vbfKP_Uj7AXotFucCFqegxebXPw-2fkJWxA-HTQDgRFw1aXPttFtwHH5mPPfmvHK5XBv9qzGia9Jv0VKKv2Bo8RjnW-DEvEOrwFTGv1M1kyILndlDP2Gxnwd4U91fWEu-tMufaLlYRe4QUS3da7Uy88OxeEWp4jxFzqkgYAzehKrr3zxmy0)

**Expected result:** I would expect activation to be the same across all participants, and preferably (IMO) like that of participant B, the first activated in the chain.

**Two related observations:** (A) Issue is the same if using ++/-- or explicit activate/deactivate statements, see example [here](http://www.plantuml.com/plantuml/umla/LOqnJWCn44NxESMdGL0a3PsBLNRN25431nZYsM39NaUUSHGuFIHYGsZVctb_eyRLMavqiw-yomoJyWMhJUWZ1i5KYhsN8rcq923FneYAAbQZ72HWgYNZ0Vgf9bdfMRM9Gx-rU62BPJvTbsbIiJNUDCuxlFuihEcklyVZSP_YDXhuSkxmmbbGHLioaZdWl4pKOzLXS7WIGs1Z-lqH37IBinqT_kYuK7zDFGrOhReElOC_LOBywumNkvYbvHTpBJnjP0ujfsy0). (B) If you put (first) activate statement before first message, you must explicitly declare A as first participant or things go nuts (not illogical, but maybe not expected/nice) and then all lifeline starts are above instead, but still with different offsets, see [here](http://www.plantuml.com/plantuml/umla/JP3FJeGm4CRlynIVbuqU837UE905QentVO3kCc0JIabdsAm-lLtYLexz_dpzPOwYDkhg9tfOeXszXNBuWSQLwSlrZ244FOKhgTE9GSQgXHD4rkZumZs663nU8D-Yx8NUHLQkq9pLNQowCD_IOHY4jSIdk7d4H_gX13ssJtYzBfCxEuND-HBFaj-C4myXygPv5h4ZbyHpZus7Q4dqQM2niw8XohMM6XGrsWflhEWJDHsmILs5_rXtLyrUDTIYAEeQJGLpg_Qy6pwWo-wVaxVCtTaRXew9EDtu5m00).

PS Related issue I found: https://forum.plantuml.net/13409/teoz-arrow-alignment. Otherwise, could not find this reported.

As always, thanks for an essential tool.
BR! /mawi

**Comment by jimnelson372** [comment]
Hello @chipbite, I mentioned on issue #739 that I was looking into this issue (and #1099), where the lifeline activity levels are incorrectly offset when parallel messages have different text heights.  In this issue, two of the parallel messages have no text and one has a single line text.   Unfortunately, I haven't been able to find a full solution to this situation yet.

The cause of the problem is that Teoz must first calculate the expected layout on each of the parallel messages before it can determine if any of them is is taller so that it can then adjust the others accordingly; however, while Teoz is adjusting the level of the message arrows themselves, it is not currently sending these adjustments to the life lines.  As a result, the life line activity levels are being drawn where their related messages would have been positioned if no adjustment was needed.

In your case, the C and D life line activations are initially calculated based on the text-less A->C and A->D messages.   The A->B message has the single line text so the B activation is calculated to be drawn lower.   Teoz detects the difference in size so adjusts the A->C and A->D arrows down, but leaves the C and D activations where they were.

I have tested changes where I pass the adjustments to the lifeline layout calculations, and your specific example seems to work.  However, if I modify the example just a little, such as moving the text from the first message to the second or third, new problems pop up.   So, I'm digging further.

BTW, my change to #739 and several other fixes I made to problems related to parallel messages have now been merged into the code.  I didn't want to hold them up while working on this issue.

Regards,

Jim Nelson

**Comment by chipbite** [comment]
Hi! Thanks for taking the time to describe the details of the issu! I am not completely following all details, will look at this a bit more later. BR! /marcus

**Comment by jimnelson372** [comment]
Quick update.  I think I found the problem I was having with my solution.   And I believe I know what to do to get this fully working.   Almost there...

**Comment by jimnelson372** [comment]
I believe I have it solved.  I'm preparing the pull request now.

**Comment by jimnelson372** [comment]
I have submitted the fix in pull request #1793.   With this fix, your example generates:

![testteozlifeline](https://github.com/plantuml/plantuml/assets/66284321/f883dca9-bbd2-43a6-93a9-f1b8687a6473)

And #1099 now generates:

![testteozlifeline_005](https://github.com/plantuml/plantuml/assets/66284321/8e607076-d472-4ed6-adfd-30d15558b8dd)

(I did not change the Notes to indicate the fix, but left it exactly as it was.)
